plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx

```diff
@@ -33,26 +36,106 @@ import {
   Progress,
   useApi,
   ResponseErrorPanel,
+  Link,
 } from '@backstage/core';
-import { Box, Typography } from '@material-ui/core';
+import { Box, makeStyles, Typography } from '@material-ui/core';
 import ZoomOutMap from '@material-ui/icons/ZoomOutMap';
 import React from 'react';
 import { useAsync } from 'react-use';
+import { BackstageTheme } from '@backstage/theme';
+import { generatePath } from 'react-router';
 
-function simplifiedEntityName(
+const useStyles = makeStyles((theme: BackstageTheme) => ({
+  domainNode: {
+    fill: theme.palette.primary.main,
+    stroke: theme.palette.border,
+  },
+  systemNode: {
+    fill: 'coral',
+    stroke: theme.palette.border,
+  },
+  componentNode: {
+    fill: 'yellowgreen',
+    stroke: theme.palette.border,
+  },
+  apiNode: {
+    fill: theme.palette.gold,
+    stroke: theme.palette.border,
+  },
+  resourceNode: {
+    fill: 'grey',
+    stroke: theme.palette.border,
+  },
+}));
+
+// Simplifies the diagram output by hiding the default namespace and kind
+function readableEntityName(
   ref:
     | Entity
     | {
-        kind?: string;
+        kind: string;
         namespace?: string;
         name: string;
       },
 ): string {
-  // Simplify the diagram output by hiding only the default namespace
-  return serializeEntityRef(ref)
-    .toString()
+  return stringifyEntityRef(ref)
     .toLocaleLowerCase('en-US')
-    .replace(':default/', ':');
+    .replace(`:${ENTITY_DEFAULT_NAMESPACE}/`, ':')
+    .split(':')[1];
+}
+
+function RenderNode(props: DependencyGraphTypes.RenderNodeProps<any>) {
+  const classes = useStyles();
+  const kind = props.node.kind || 'Component';
+  const ref = parseEntityRef(props.node.id);
+  let nodeClass = classes.componentNode;
+
+  switch (kind) {
+    case 'Domain':
+      nodeClass = classes.domainNode;
+      break;
+    case 'System':
+      nodeClass = classes.systemNode;
+      break;
+    case 'Component':
+      nodeClass = classes.componentNode;
+      break;
+    case 'API':
+      nodeClass = classes.apiNode;
+      break;
+    case 'Resource':
+      nodeClass = classes.resourceNode;
+      break;
+    default:
+      nodeClass = classes.componentNode;
+  }
+
+  return (
+    <g>
+      <rect width={200} height={100} rx={20} className={nodeClass} />
+      <Link
+        to={generatePath(`/catalog/${entityRoute.path}`, {
+          kind: kind,
+          namespace: ref.namespace,
+          name: ref.name,
+        })}
```

Comment:
Although it will work most of the times, I feel it's not okay for this component to assume the `/catalog` endpoint for Software Catalog.

How about using the `useRouteRef(catalogEntityRouteRef)` to generate the URL for the entity? An example -
https://github.com/backstage/backstage/blob/2f423757d2f14eec597bb77651f5b5825fa341c0/plugins/explore/src/components/DomainCard/DomainCard.tsx#L39-L42

Reply:
Good catch :+1: 

I updated the PR.